public/javascripts/registration-validation.js

```diff
@@ -0,0 +1,61 @@
+var errorNames = [];
+var validator = new FormValidator('registration_form', [{
+  name: 'firstname',
+  display: 'required',
+  rules: 'required|alpha'
+},
+  {
+    name: 'lastname',
+    display: 'required',
+    rules: 'required|alpha'
+  },
+  {
+    name: 'password',
+    rules: 'required|callback_check_password'
+  },
+  {
+    name: 'repeatpassword',
+    display: 'repeated password',
+    rules: 'required|matches[password]'
+  },
+  {
+    name: 'email',
+    rules: 'required|valid_email'
+  },
+  {
+    name: 'terms',
+    display: 'terms &amp; conditions',
+    rules: 'required'
+  }
+], function(errors, event) {
+  for(var j=0; j<errorNames.length; j++){
+    document.getElementById(errorNames[j]+'_error').setAttribute('style','display:none;');
+  }
+  errorNames = [];
+  if (errors.length > 0) {
+    event.preventDefault();
+    for(var i=0; i<errors.length; i++){
+      var error = errors[i];
+      error.element.setAttribute('style','border: solid 1px red;')
+      if(error.rule==='required'&&error.name!='terms'){
+        error.element.setAttribute('placeholder', 'This field is required.');
+      }
+      else{
+        errorNames.push(error.name);
+        document.getElementById(error.name+'_error').setAttribute('style','display:block;');
+      }
+    }
+  };
+});
+validator.registerCallback('check_password', function(value) {
+  //At least one upper case english letter
+  //At least one lower case english letter
+  //At least one digit
+  //At least one special character
+  //Minimum 8 in length
```

Comment:
Do we have these password requirements? 

Also, your definition of "special characters" is rather small. for example, the backtick, or `'`, or `"`, or `;` or `é`, or `ü` are special characters, too, but they are not in the list. 

Further, validation should happen on the backend - otherwise someone can simply disable javascript (or change it) and get around this validation. There is the password validatior class that can be extended to have an own `validate` method, validation could happen there. See [https://github.com/jaliss/securesocial/blob/db46f2c3db784f3596ddd407075c222952d08129/module-code/app/securesocial/core/providers/utils/PasswordValidator.scala](https://github.com/jaliss/securesocial/blob/db46f2c3db784f3596ddd407075c222952d08129/module-code/app/securesocial/core/providers/utils/PasswordValidator.scala)